app/models/post.rb

```diff
@@ -1,3 +1,17 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+  after_create :increment_user_posts_counter
+  def increment_user_posts_counter
+    author.increment!(:posts_counter)
```

Comment:
- [optional] you can also use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private. you can also use this for the comment and like model methods
eg

```ruby
after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```